packages/fxa-settings/src/pages/Index/container.tsx

```diff
@@ -158,36 +172,83 @@ export const IndexContainer = ({
         return getHandledError(error);
       }
     },
-    [authClient, handleSuccess, isWebChannelIntegration]
+    [authClient, handleSuccess, integration]
   );
 
+  // If there is a suggested email, try to direct to either signin or signup
+  // If no account exists for the suggested email, the same email validation steps
+  // will be run as if the email was submitted manually.
+  // RP-provided 'email' query param should take precedence, followed by 'login_hint'
+  // then previous accounts.
+  const suggestedEmail =
+    queryParamModel.email ||
+    queryParamModel.loginHint ||
+    currentAccount()?.email ||
+    lastStoredAccount()?.email;
+
+  // If we just came from another Mozilla accounts page with a prefill email in location state,
+  // ignore suggested email (they are probably stale)
+  const useEmailSuggestion = suggestedEmail && !prefillEmail;
+
   useEffect(() => {
-    if (!hasEmailSuggestion) {
+    if (!useEmailSuggestion) {
       return;
     }
 
-    signUpOrSignInHandler(suggestedEmail, false);
-  }, [hasEmailSuggestion, signUpOrSignInHandler, suggestedEmail]);
+    (async () => {
+      const { error } = await signUpOrSignInHandler(suggestedEmail, false);
+      if (error) {
+        const localizedError = getLocalizedEmailValidationErrorMessage(
+          error,
+          ftlMsgResolver,
+          suggestedEmail
+        );
+        switch (error.errno) {
```

Comment:
We have the same logic in Index.tsx repeated. To me the page should decide it's error banners and tooltips, since these things are UX based. 

Since the prefilled email is rendered into the page, perhaps this logic belongs in the index.tsx (the presentation layer) and the container?

Reply:
Updated to remove duplication and only pass in the error messages to the presentation component.